src/services/location.js

```diff
@@ -0,0 +1,18 @@
+export const GetLocation = () => {
+    return new Promise((resolve, reject) => {
+        if (navigator.geolocation) {
+            navigator.geolocation.getCurrentPosition(
+                (position) => {
+                    const { latitude, longitude } = position.coords;
+                    resolve({ latitude, longitude });
+                },
+                (error) => {
+                    console.error(error);
+                    reject(new Error("Failed to retrieve location."));
+                }
+            );
+        }
```

Comment:
what do you think should happen if there's no `navigator.geolocation`? Shall we handle this situation somehow too?